src/applicationManager.test.ts

```diff
@@ -142,11 +159,39 @@ describe("applicationManager", () => {
       optionalClaims: [{ name: "groups", additionalProperties: [] }],
     });
 
-    const application = await readApplication({
+    let application = await readApplication({
+      token,
+      applicationId: appDetails.applicationId,
+    });
+
+    const groupId = await getEntraGroupId(appRoles[0].groups[0], token);
+
+    await addAppRoles({
+      token,
+      applicationId: appDetails.applicationId,
+      appRoles: appRoles,
+    });
+
+    await addAppRoleGroupAssignmentsToApp({
+      token: token,
+      applicationId: appDetails.servicePrincipalObjectId,
+      appRoles: appRoles,
+    });
+
+    // Needs to be re-read after updating
+    application = await readApplication({
```

Comment:
have you run this a few times to make sure it's reliable? graph APIs are generally eventually consistent so may take some time for the new values to be present.

Reply:
Yes - there's sometimes errors during the test (~10%) but only when the cleanup is going on - so will take a look into this further

Reply:
It only seems to be the setLogo test failing now and then:
![image](https://github.com/hmcts/azure-app-proxy-manager/assets/47995122/5e93f0aa-326e-4c43-a810-9197176f95f5)

Will look into it